src/lib.rs

```diff
@@ -62,17 +70,24 @@ pub type Result<T, E = CoreError> = std::result::Result<T, E>;
 /// expected that only one instance of an implementation will exist for the lifetime of the
 /// worker(s) using it.
 pub trait Core: Send + Sync {
-    /// Ask the core for some work, returning a [Task], which will eventually contain either a
-    /// [protos::coresdk::WfActivation] or an [protos::coresdk::ActivityTask]. It is then the
-    /// language SDK's responsibility to call the appropriate code with the provided inputs.
+    /// Ask the core for some work, returning a [Task], which will contain a [protos::coresdk::WfActivation].
+    /// It is then the language SDK's responsibility to call the appropriate code with the provided inputs.
     ///
     /// TODO: Examples
+    /// TODO: rename to poll_workflow_task and change result type to WfActivation
     fn poll_task(&self, task_queue: &str) -> Result<Task>;
 
+    /// Ask the core for some work, returning a [protos::coresdk::Task], which will contain a [protos::coresdk::ActivityTask].
```

Comment:
We should get rid of top-level Task -- though we can do that in the "refactor proto" PR if desired

Reply:
You might wanna do it now though just because it removes a lot of pointless matching in this PR... up to you

Reply:
Makes sense, we can move task token inside of each result type.

Reply:
Let's do it in a follow up, there are a few things we should discuss, so it's more of a stand-alone refactor.